### ISSUE TITLE
sysmonitor@orcus fails to load with Applet.AllowedLayout is undefined

### ISSUE THREAD

**Issue by jbaldassari** [linuxmint/cinnamon-spices-applets#1357]
* Applet version/Build date (to be found in "About.." or in "metadata.json")
```
{
    "last-edited": 1507062205, 
    "max-instances": -1, 
    "dangerous": true, 
    "uuid": "sysmonitor@orcus", 
    "name": "System Monitor", 
    "version": "1.6.1", 
    "icon": "utilities-system-monitor", 
    "description": "Displays CPU, memory, swap and network usage and load in graphs"
}
```

 * Cinnamon version (cinnamon --version)
```
Cinnamon 3.0.7
```

 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
```
Gentoo
```

 * Graphics hardware *and* driver used
```
Hardware: Intel Corporation HD Graphics 520 (rev 07)
Driver: i915
```

 * Snippet or attachment of ~/.cinnamon/glass.log
```
error t=2017-10-04T19:20:41.681Z Applet.AllowedLayout is undefined
trace t=2017-10-04T19:20:41.682Z 
<----------------
MyApplet.prototype._init@/home/james/.local/share/cinnamon/applets/sysmonitor@orcus/applet.js:69
MyApplet@/home/james/.local/share/cinnamon/applets/sysmonitor@orcus/applet.js:60
main@/home/james/.local/share/cinnamon/applets/sysmonitor@orcus/applet.js:364
createApplet@/usr/share/cinnamon/js/ui/appletManager.js:329
addAppletToPanels@/usr/share/cinnamon/js/ui/appletManager.js:256
finishExtensionLoad@/usr/share/cinnamon/js/ui/appletManager.js:59
loadExtension@/usr/share/cinnamon/js/ui/extension.js:451
onEnabledAppletsChanged@/usr/share/cinnamon/js/ui/appletManager.js:197
---------------->
error t=2017-10-04T19:20:41.682Z [Applet "sysmonitor@orcus"]: Failed to evaluate 'main' function on applet: sysmonitor@orcus/35
error t=2017-10-04T19:20:41.682Z Could not load applet sysmonitor@orcus
```

 * Snippet or attachment of ~/.xsession-errors
N/A

**Notify author of applet**
@OrcusCZ 

**Issue**

After updating all applets today the ```sysmonitor@orcus``` applet fails to load.  The following messages were in my Cinnamon log:

```
error t=2017-10-04T19:20:41.681Z Applet.AllowedLayout is undefined
Failed to evaluate 'main' function on applet: sysmonitor@orcus/35
```

**Steps to reproduce**
Add applet to panel

**Expected behaviour**
Applet loads normally

**Other information**

**Comment by NikoKrause** [comment]
It's because the `AllowedLayout` was introduced in Cinnamon 3.2 for vertical panels.

@OrcusCZ 
You can fix this by using `try/catch` as it was done here: https://github.com/linuxmint/cinnamon-spices-applets/pull/1242/commits/9875bbce97b912d8cecc4d43425a7375e7f097c6

**Comment by jbaldassari** [comment]
Thanks.  I tried the modification you suggested locally, but the applet wouldn't load for a different reason.  I think there are probably other things that are not compatible with pre-3.2 Cinnamon in here.  If @OrcusCZ can figure out the issues that's great, but I might just try to get my Cinnamon upgraded to a newer version.

**Comment by OrcusCZ** [comment]
I've installed your Cinnamon version to a virtual machine, fixed that AllowedLayout issue and found out why it won't load. I use layouts and switch controls in settings schema, they have been added in Cinnamon 3.2. To make the settings schema work in 3.0.7, layouts must be removed and switches changed to checkboxes (which are deprecated since 3.2).

I can make the applet multiversion to support both schemas, I don't think there is a better solution (correct me if I am wrong). I just tried it in my VM, it works for me and upgrading Cinnamon from 3.0.7 to 3.4.6 (version I use now) worked well and kept the settings.

@jbaldassari can you try my updated version and tell me if it works for you?
Clone branch sysmon-3.0.7 from my fork, for ex. using:
`git clone -b sysmon-3.0.7 https://github.com/OrcusCZ/cinnamon-spices-applets`
and copy the folder `cinnamon-spices-applets/sysmonitor@orcus/files/sysmonitor@orcus` to `~/.local/share/cinnamon/applets/` after removing the applet you currently have installed. Then restart Cinnamon and it should work.

**Comment by jbaldassari** [comment]
Hi @OrcusCZ.  Thanks for the quick reply.  I actually upgraded by Cinnamon to 3.2 this morning as a workaround, and then the applet worked perfectly without any modifications.  I'm sorry to have made you go through that effort, but maybe others who are still on 3.0 will benefit from the fix.  Thanks for your excellent work on this applet.  It's still my favorite system monitor!

**Comment by OrcusCZ** [comment]
Hi, no problem. I've a VM setup for this because every time someone has a problem, it's always different version of Cinnamon than mine. I'll assume that this fixes the issue for everyone on 3.0, since it works for me. Glad you like the applet :)
